fabric-input-events-v1/src/main/java/net/fabricmc/fabric/api/event/client/input/ClientInputEvents.java

```diff
@@ -0,0 +1,160 @@
+			listener.onKey(key);
+		}
+	});
+	/**
+	 * Called when the player presses a key that is bound to some keybind.
+	 */
+	public static final Event<KeybindState> KEYBIND_PRESSED = EventFactory.createArrayBacked(KeybindState.class, listeners -> key -> {
+		for (KeybindState listener : listeners) {
+			listener.onKeybind(key);
+		}
+	});
+	/**
+	 * Called when the player releases a key that is bound to some keybind.
+	 */
+	public static final Event<KeybindState> KEYBIND_RELEASED = EventFactory.createArrayBacked(KeybindState.class, listeners -> key -> {
+		for (KeybindState listener : listeners) {
+			listener.onKeybind(key);
+		}
+	});
+	/**
+	 * Called when the player presses a key that is bound to some keybind for a while.
+	 */
+	public static final Event<KeybindState> KEYBIND_REPEATED = EventFactory.createArrayBacked(KeybindState.class, listeners -> key -> {
+		for (KeybindState listener : listeners) {
+			listener.onKeybind(key);
+		}
+	});
+	/**
+	 * Called when the player types a character in some way.
+	 *
+	 * <p>CHAR_TYPED events need not have a corresponding KEY_PRESSED event,
+	 * for example when pressing dead keys or using an Emoji menu.
+	 */
+	public static final Event<CharState> CHAR_TYPED = EventFactory.createArrayBacked(CharState.class, listeners -> (codepoint, modKeys) -> {
+		for (CharState listener : listeners) {
+			listener.onChar(codepoint, modKeys);
+		}
+	});
+	/**
+	 * Called when the player moves their mouse.
+	 */
+	public static final Event<MouseMove> MOUSE_MOVED = EventFactory.createArrayBacked(MouseMove.class, listeners -> mouse -> {
+		for (MouseMove listener : listeners) {
+			listener.onMouseMoved(mouse);
+		}
+	});
+	/**
+	 * Called when the player presses a button on their mouse.
+	 */
+	public static final Event<MouseButtonState> MOUSE_BUTTON_PRESSED = EventFactory.createArrayBacked(MouseButtonState.class, listeners -> mouse -> {
+		for (MouseButtonState listener : listeners) {
+			listener.onMouseButton(mouse);
+		}
+	});
+	/**
+	 * Called when the player releases a button on their mouse.
+	 */
+	public static final Event<MouseButtonState> MOUSE_BUTTON_RELEASED = EventFactory.createArrayBacked(MouseButtonState.class, listeners -> mouse -> {
+		for (MouseButtonState listener : listeners) {
+			listener.onMouseButton(mouse);
+		}
+	});
+	/**
+	 * Called when the player scrolls their mouse wheel.
+	 */
+	public static final Event<MouseScroll> MOUSE_WHEEL_SCROLLED = EventFactory.createArrayBacked(MouseScroll.class, listeners -> mouse -> {
+		for (MouseScroll listener : listeners) {
+			listener.onMouseScrolled(mouse);
+		}
+	});
+	/**
+	 * Called when the player drops a file into the Minecraft window.
+	 */
+	public static final Event<FileDrop> FILE_DROPPED = EventFactory.createArrayBacked(FileDrop.class, listeners -> paths -> {
+		for (FileDrop listener : listeners) {
+			listener.onFilesDropped(paths);
+		}
+	});
+
+	@FunctionalInterface
+	public interface KeyState {
+		void onKey(KeyEvent key);
```

Comment:
To clarify, all the event objects just not char

Reply:
So we are doing 6 parameter listeners? Doesn't that seem extremely inconvenient?